src/openms/source/ANALYSIS/ID/AScore.cpp

```diff
@@ -100,25 +114,29 @@ namespace OpenMS
         if (permu == permutations[i].size())
           break;
       }
-      spectrum_generator.addPeaks(th_spectra[i], temp, Residue::BIon, hit.getCharge());
-      spectrum_generator.addPeaks(th_spectra[i], temp, Residue::YIon, hit.getCharge());
+
+      // we mono-charge spectra
+      spectrum_generator.addPeaks(th_spectra[i], temp, Residue::BIon, 1);
+      spectrum_generator.addPeaks(th_spectra[i], temp, Residue::YIon, 1);
       th_spectra[i].setName(temp.toString());
     }
-    ///produce theoretical spectra - END
 
     if (!real_spectrum.isSorted())
     {
       real_spectrum.sortByPosition();
     }
+
     vector<vector<double> > peptide_site_scores(th_spectra.size());
-    vector<RichPeakSpectrum> windows_with_all_peak_depths;
-    ///prepare peak depth for all windows in the actual spectrum
+    vector<RichPeakSpectrum> windows_top10;
+
+    // prepare peak depth for all windows in the actual spectrum
     {
-      double biggest_window = real_spectrum[real_spectrum.size() - 1].getMZ();
-      Size number_of_windows = 1;
-      if (biggest_window > 100)
-        number_of_windows = ceil(biggest_window / 100);
-      windows_with_all_peak_depths.resize(number_of_windows);
+      double biggest_window = real_spectrum.back().getMZ();
+      Size number_of_windows = static_cast<Size>(ceil(biggest_window / 100) + 0.5);
```

Comment:
Are you sure this is what you want?
ceil(x) + 0.5 looks like its always == ceil(x)